### ISSUE TITLE
Should pthreads be a public dependency?

### ISSUE THREAD

**Issue by chausner** [David-Haim/concurrencpp#77]
In https://github.com/David-Haim/concurrencpp/blob/580f430caf1369521ee1422c15c4c8a89823972b/cmake/coroutineOptions.cmake, pthreads is a `PRIVATE` dependency:
```cmake
target_link_libraries(${TARGET} PRIVATE Threads::Threads)
```

Not only the main library but also the unit tests call the `target_coroutine_options` CMake function, thus the tests link to pthreads. However, when I try to build this simple example on Linux, I get a linker error missing the pthreads library if I don't explicitly link to pthreads:
```c++
#include "concurrencpp/concurrencpp.h"
#include <iostream>

int main() {
    concurrencpp::runtime runtime;
    auto result = runtime.thread_executor()->submit([] {
        std::cout << "hello world" << std::endl;
    });

    result.get();
    return 0;
}
```
```cmake
find_package(concurrencpp REQUIRED CONFIG)

add_executable(${PROJECT_NAME} test_package.cpp)
target_link_libraries(${PROJECT_NAME} concurrencpp::concurrencpp)
target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_20)
```

Should `Threads::Threads` be made a `PUBLIC` dependency so that any consumer links to pthreads automatically? Or is it intentional that users must link to pthreads manually? Is there any  reasonable way to use the library without pthreads?

**Comment by David-Haim** [comment]
Good question.
@friendlyanon worked on our CMake some time ago and made it so. 
I have no problem with making pthreads public, if it makes the linking easier for consumers. 

> Is there any reasonable way to use the library without pthreads?

You could, theoretically, only use `lazy_result` and `generator` which don't use any multi-threading primitive at all. 
However you are right that this library is meant to be used in a multi-threaded environment and I fail to see any sane way to use it without `pthread`

If you want to experiment with this configuration and make a PR feel free to do so,
Thanks!

**Comment by friendlyanon** [comment]
That contribution was made when I knew way less about build and usage requirements, and how to best express them in CMake.

Basically, any dependency you link to, must also be propagated using `find_dependency`, be it an internal implementation detail or otherwise, for properly defined transitive requirements.  
You can find a small example showing this here: https://github.com/friendlyanon/cmake-init-transitive

You may also find improvements in general regarding project CMake code in cmake-init, which the above example is based on.

**Comment by chausner** [comment]
> Basically, any dependency you link to, must also be propagated using find_dependency, be it an internal implementation detail or otherwise, for properly defined transitive requirements.

Thanks a lot for that hint, this is really helpful information. Added `find_dependency` in https://github.com/David-Haim/concurrencpp/pull/78/commits/3186c35f7c5e5a3ad387bd7878e705649fed5834 to my PR #78.

**Comment by friendlyanon** [comment]
Truth be told, the build files could use some general cleaning up. That `rt` business is gnarly in particular.

**Comment by David-Haim** [comment]
Solved with #78 .
Thanks @chausner .
@friendlyanon if you feel like you want to overhaul the build files - go for it.
